lib/src/request.dart

```diff
@@ -30,12 +30,12 @@ class Request extends Message {
   ///
   /// Extra [context] can be used to pass information between inner middleware
   /// and handlers.
-  Request(this.method, this.url,
+  Request(String method, url,
```

Comment:
Document the acceptable types for `url`. You can probably just copy the docs from the top-level methods.